examples/celllistmap.jl

```diff
@@ -0,0 +1,249 @@
+# simulation. The `Particle` type, for the `ContinousAgent{2}` space, will have additionally
+# an `id` and `pos` (positon) and `vel` (velocity) fields, which are automatically added
+# by the `@agent` macro.
+@agent Particle ContinuousAgent{2} begin
+    r::Float64 # radius
+    k::Float64 # repulsion force constant
+    mass::Float64
+end
+Particle(; id, pos, vel, r, k, mass) = Particle(id, pos, vel, r, k, mass)
+
+# ## Required and data structures for CellListMap.jl
+import CellListMap
+using StaticArrays
+# `StaticArrays` provides the `SVector` type, which is practical for the representation of
+# various vector types (e.g., coordinates or velocities) in small amount of dimensions.
+# Agents.jl uses `NTuple{D, Float64}` for that, which does not support vector operations
+# out of the box. In the future, Agents.jl may also switch the `pos` type to a static vector.
+
+# We `import` the `CellListMap` package because
+# it exports some functions with conflicting names with `Agents`.
+
+# 
+# ## Model properties.
+# 
+# The `clmap_system` field will store the data required for `CellListMap`, and again we 
+# use a parametric type to guarantee the concrectness of the types, aoviding type
+# instabilities.  
+#
+Base.@kwdef struct Properties{CL<:CellListMap.PeriodicSystem}
+    dt::Float64 = 0.01
+    number_of_particles::Int64 = 0
+    clmap_system::CL # CellListMap system
+end
+
+# Two auxiliary arrays will be created on model initialization, to be passed to
+# the `CellListMap.PeriodicSystem` data structure:
+#
+# 1. `positions`: `CellListMap` requires a vector of (preferentially) static vectors as the positions
+#    of the particles. To avoid creating this array on every call, a buffer to
+#    which the `agent.pos` positions will be copied is stored in this data structure.
+# 2. `forces`: In this example, the property to be computed using `CellListMap.jl` is 
+#    the forces between particles, which are stored here in a `Vector{<:SVector}`, of
+#    the same type as the positions. These forces will be updated by the `CellListMap.map_pairwise!`
+#    function.
+#
+# Additionally, the computation with `CellListMap.jl` requires the definition of a `cutoff`,
+# which will be twice the maximum interacting radii of the particles, and the geometry of the
+# the system, given by the `sides` of the periodic box. 
+#
+# Just for clarity, we define an alias for the function that fetches the forces from 
+# the `CellListMap.PeriodicSystem` structure:
+const get_forces = CellListMap.get_output
+
+#
+# ## Model initialization
+# We create the model with a keyword-accepting function as is recommended in Agents.jl.
+# The keywords here control number of particles and sizes.
+function initialize_model(;
+    number_of_particles=10_000,
+    sides=SVector(500.0, 500.0),
+    dt=0.001,
+    max_radius=10.0,
+    parallel=true
+)
+    ## initial random coordinates
+    coordinates = [sides .* rand(SVector{2,Float64}) for _ in 1:number_of_particles]
```

Comment:
Hm, I find the name `coordinates` confusing, I think `positions` is much better. The latter more clearly reflects that this is the positions of all particles. Coordinates, even if plural, may mean the coordinate system. At first I though that this variable was about the internal cell size of the cell list.